pkg/sql/backfill.go

```diff
@@ -802,7 +802,11 @@ func TruncateInterleavedIndexes(
 		// All the data chunks have been removed. Now also removed the
 		// zone configs for the dropped indexes, if any.
 		if err := db.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
-			return RemoveIndexZoneConfigs(ctx, txn, execCfg, table.GetParentID(), indexes)
+			freshTableDesc, err := catalogkv.MustGetTableDescByID(ctx, txn, execCfg.Codec, table.GetID())
+			if err != nil {
+				return err
+			}
+			return RemoveIndexZoneConfigs(ctx, txn, execCfg, freshTableDesc, indexes)
```

Comment:
I changed the signature of `RemoveIndexZoneConfigs` to take a `catalog.TableDescriptor` because otherwise it loads it from storage using a catalogkv api call which causes cross-reference validation check failures in `indexTruncateInTxn`.
This is another of these new failures which cropped up after I raised the validation level for mutable descs.